### ISSUE TITLE
Issues Importing tensorflow.contrib

### ISSUE THREAD

**Issue by calvinmccormack** [Elvenson/midiMe#18]
Hi,

I've been running into some issues trying to run these .py files. It keeps returning an error whenever it tries to import tensorflow.contrib: "ModuleNotFoundError: No module named 'tensorflow.contrib'", since contrib was sunset in Tensorflow 2.0.  

Reverting back to Tensorflow 1.14 was also unsuccessful since it causes problems with a different import: "ImportError: This version of TensorFlow Datasets requires TensorFlow version >= 2.1.0"

Might there be some way around this or a way to upgrade the py. files happily? 

Thank you!

**Comment by Elvenson** [comment]
Hi there thanks for raising an issue, I think you can solve this by creating new python environment and type:
`pip install -r requirement.txt`
As for `requirement.txt` file, you can create it and put the following data inside:
```
absl-py==0.9.0
apache-beam==2.18.0
appnope==0.1.0
astor==0.8.1
astroid==2.4.2
attrs==19.3.0
audioread==2.1.8
avro-python3==1.9.2.1
backcall==0.1.0
backports.tempfile==1.0
backports.weakref==1.0.post1
bleach==3.1.4
bokeh==2.0.1
bz2file==0.98
cachetools==3.1.1
certifi==2020.12.5
cffi==1.14.0
chardet==3.0.4
click==7.1.1
cloudpickle==1.2.2
contextlib2==0.6.0.post1
crcmod==1.7
cycler==0.10.0
dataclasses==0.6
decorator==4.4.2
defusedxml==0.6.0
dgl==0.4.3
dglke==0.1.1
dill==0.3.1.1
dm-sonnet==1.35
docopt==0.6.2
dopamine-rl==3.0.1
entrypoints==0.3
faiss==1.6.1
fastavro==0.21.24
fasteners==0.15
Flask==1.1.2
future==0.18.2
gast==0.2.2
gevent==1.5.0
gin-config==0.3.0
google-api-core==1.17.0
google-api-python-client==1.8.0
google-apitools==0.5.28
google-auth==1.14.0
google-auth-httplib2==0.0.3
google-cloud-bigquery==1.17.1
google-cloud-bigtable==1.0.0
google-cloud-core==1.3.0
google-cloud-datastore==1.7.4
google-cloud-pubsub==1.0.2
google-pasta==0.2.0
google-resumable-media==0.4.1
googleapis-common-protos==1.51.0
greenlet==0.4.15
grpc-google-iam-v1==0.12.3
grpcio==1.28.1
gunicorn==20.0.4
gym==0.15.7
h5py==2.10.0
hdfs==2.5.8
httplib2==0.12.0
idna==2.9
imageio==2.8.0
importlib-metadata==1.6.0
intervaltree==3.0.2
ipykernel==5.2.1
ipython==7.13.0
ipython-genutils==0.2.0
ipywidgets==7.5.1
isort==4.3.21
itsdangerous==1.1.0
jedi==0.17.0
Jinja2==2.11.2
joblib==0.14.1
jsonschema==3.2.0
jupyter==1.0.0
jupyter-client==6.1.3
jupyter-console==6.1.0
jupyter-core==4.6.3
Keras-Applications==1.0.8
Keras-Preprocessing==1.1.0
kfac==0.2.0
kiwisolver==1.2.0
lazy-object-proxy==1.4.3
librosa==0.7.2
lightgbm==2.2.2
llvmlite==0.32.0
magenta==1.3.1
Markdown==3.2.1
MarkupSafe==1.1.1
matplotlib==3.2.1
mccabe==0.6.1
mesh-tensorflow==0.1.13
mido==1.2.6
mir-eval==0.6
mistune==0.8.4
mkl-fft==1.0.15
mkl-random==1.1.0
mkl-service==2.3.0
mock==2.0.0
monotonic==1.5
more-itertools==8.3.0
mpmath==1.1.0
nbconvert==5.6.1
nbformat==5.0.6
networkx==2.4
notebook==6.0.3
numba==0.49.0
numpy @ file:///opt/concourse/worker/volumes/live/a1b5ea96-b05c-40ba-5dfb-4542ece1c186/volume/numpy_and_numpy_base_1603491228242/work
oauth2client==3.0.0
opencv-python==4.2.0.34
opt-einsum==3.2.1
packaging==20.3
pandas==1.0.3
pandocfilters==1.4.2
parso==0.7.0
pbr==5.4.5
pexpect==4.8.0
pickleshare==0.7.5
Pillow==7.1.1
pluggy==0.13.1
pretty-midi==0.2.9
prometheus-client==0.7.1
promise==2.3
prompt-toolkit==3.0.5
protobuf==3.11.3
ptyprocess==0.6.0
py==1.8.1
pyarrow==0.15.1
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycairo==1.19.1
pycparser==2.20
pydot==1.4.1
pyglet==1.5.0
Pygments==2.6.1
pygtrie==2.3.3
pylint==2.5.3
pymongo==3.10.1
pyparsing==2.4.7
pypng==0.0.20
pyrsistent==0.16.0
pytest==5.4.3
pytest-pylint==0.17.0
python-dateutil==2.8.1
python-rtmidi==1.1.2
pytz==2019.3
PyWavelets==1.1.1
PyYAML==5.3.1
pyzmq==19.0.0
qtconsole==4.7.3
QtPy==1.9.0
requests==2.23.0
resampy==0.2.2
rsa==4.0
scikit-image==0.16.2
scikit-learn==0.22.2.post1
scipy==1.4.1
seaborn==0.10.1
semantic-version==2.8.4
Send2Trash==1.5.0
shopee-mesos-cli==1.0.58
six @ file:///opt/concourse/worker/volumes/live/f983ba11-c9fe-4dff-7ce7-d89b95b09771/volume/six_1605205318156/work
sk-video==1.1.10
sortedcontainers==2.1.0
SoundFile==0.10.3.post1
sox==1.3.7
sympy==1.5.1
tensor2tensor==1.15.5
tensorboard==1.15.0
tensorflow==1.15.2
tensorflow-datasets==3.0.0
tensorflow-estimator==1.15.1
tensorflow-gan==2.0.0
tensorflow-hub==0.8.0
tensorflow-metadata==0.21.2
tensorflow-probability==0.7.0
termcolor==1.1.0
terminado==0.8.3
testpath==0.4.4
toml==0.10.1
torch==1.7.0
torchbiggraph==1.0.1.dev0
tornado==6.0.4
tqdm==4.45.0
traitlets==4.3.3
typed-ast==1.4.1
typing-extensions==3.7.4.2
uritemplate==3.0.1
urllib3==1.25.9
wcwidth==0.1.9
webencodings==0.5.1
Werkzeug==1.0.1
widgetsnbextension==3.5.1
wrapt==1.12.1
xgboost==1.2.0
xgboost-deploy==0.0.2
zipp==3.1.0
```

**Comment by Elvenson** [comment]
I also just updated the README.md installation guide hope this help

**Comment by calvinmccormack** [comment]
Hey, thanks for including this requirement list

So I ended up having to modify this a bit, numpy and six appear to point to local files, and there were a few that pip couldn't find or whose version had a dependency conflict, but none of them seemed relevant to this project (hopefully).  Also, the environment had to be Python 3.7 (3.8 refused to install Tensorflow 1.15.2).  At the end of the day, here's the `requirement.txt` I ended up with: 

[requirement.txt](https://github.com/Elvenson/midiMe/files/6015406/requirement.txt)

And it appears that all is well now, at least with regards to the environment and the legacy Tensorflow modules.  I'm going to work through it and see if any other issues arise down the road, but looking good so far, thanks again!
